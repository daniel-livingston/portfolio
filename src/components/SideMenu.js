import React from "react";
import { NavLink } from "react-router-dom";
import PropTypes from "prop-types";

const propTypes = {
	closeMenu: PropTypes.func.isRequired,
};

const SideMenu = (props) => {
	const onClick = () => {
		props.closeMenu();
	};
	return (
		<div id='side-menu' className='side-menu'>
			<NavLink onClick={onClick} className='navlink' to='/' activeClassName='navlink__active'>
				About
			</NavLink>
			<NavLink
				onClick={onClick}
				className='navlink'
				to='/projects'
				activeClassName='navlink__active'
			>
				Projects
			</NavLink>
			<NavLink
				onClick={onClick}
				className='navlink'
				to='/contact'
				activeClassName='navlink__active'
			>
				Contact
			</NavLink>
		</div>
	);
};

SideMenu.propTypes = propTypes;

export default SideMenu;
