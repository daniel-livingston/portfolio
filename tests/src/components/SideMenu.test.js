import React from "react";
import { mount, shallow } from "enzyme";
import { NavLink, MemoryRouter } from "react-router-dom";
import SideMenu from "../../../src/components/SideMenu";

let closeMenu;

beforeEach(() => {
	closeMenu = jest.fn();
});

describe("<SideMenu />", () => {
	test("renders snapshot correctly with 3 NavLink components", () => {
		const wrapper = shallow(<SideMenu closeMenu={closeMenu} />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find(NavLink)).toHaveLength(3);
	});

	test("clicking NavLink closes menu", () => {
		const wrapper = mount(
			<MemoryRouter>
				<SideMenu closeMenu={closeMenu} />
			</MemoryRouter>
		);

		wrapper.find(NavLink).at(0).simulate("click");
		expect(closeMenu).toHaveBeenCalledTimes(1);
		wrapper.find(NavLink).at(1).simulate("click");
		expect(closeMenu).toHaveBeenCalledTimes(2);
		wrapper.find(NavLink).at(2).simulate("click");
		expect(closeMenu).toHaveBeenCalledTimes(3);
	});
});
