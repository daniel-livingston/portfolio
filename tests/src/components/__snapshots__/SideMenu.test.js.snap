// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideMenu /> renders snapshot correctly with 3 NavLink components 1`] = `
<div
  className="side-menu"
  id="side-menu"
>
  <NavLink
    activeClassName="navlink__active"
    className="navlink"
    onClick={[Function]}
    to="/about"
  >
    About
  </NavLink>
  <NavLink
    activeClassName="navlink__active"
    className="navlink"
    onClick={[Function]}
    to="/projects"
  >
    Projects
  </NavLink>
  <NavLink
    activeClassName="navlink__active"
    className="navlink"
    onClick={[Function]}
    to="/contact"
  >
    Contact
  </NavLink>
</div>
`;
